webserver/__init__.py

```diff
@@ -20,24 +31,37 @@ def create_app_with_configuration(config_path=None):
     )
 
     # Configuration
-    root_path = os.path.join(os.path.dirname(os.path.realpath(__file__)), "..")
-    app.config.from_pyfile(os.path.join(
-        root_path,
-        'default_config.py'
-    ))
-    app.config.from_pyfile(os.path.join(
-        root_path,
-        'custom_config.py'
-    ), silent=True)
+
+    # if deploying in a production docker environment, load consul_config.py
+    if deploy_env and not config_path:
+        consul_config = os.path.join(os.path.dirname(os.path.realpath(__file__)), "..", "consul_config.py")
+        print("Checking if consul generated config file exists: %s" % consul_config)
+        for _ in range(CONSUL_CONFIG_FILE_RETRY_COUNT):
+            if not os.path.exists(consul_config):
+                time.sleep(1)
+
+        if not os.path.exists(consul_config):
+            print("No config file generated. Retried %d times, exiting." % CONSUL_CONFIG_FILE_RETRY_COUNT)
+            sys.exit(-1)
+
+        print('Loading consul_config.py...')
+        app.config.from_pyfile(consul_config)
+        return app
+
+    print('Loading config.py...')
+    config_file = os.path.join(os.path.dirname(os.path.realpath(__file__)), "..", "config.py")
+    app.config.from_pyfile(config_file)
 
     if config_path:
-        app.config.from_pyfile(config_path)
+        app.config.from_pyfile(config_path, silent=True)
```

Comment:
This has some awkward behaviour which is causing me some problems when running tests:

* the `config_path` argument was to specify a separate config file for testing. If we no longer call this method with the parameter then we should remove it
* In ListenBrainz you rename `consul_config.py.tmpl` to `config.py` (https://github.com/metabrainz/listenbrainz-server/blob/8d12052fc15c6a0a9349b51a9f53e7f1020b4485/docker/consul-template.conf#L3), but in AcousticBrainz you rename it to `consul_config.py`, which is inconsistent
* config.py is loaded unconditionally, which will probably cause a server startup error on production, as the config file is consul_config.py
* If config.py exists, this will overwrite the production config
* Unconditionally prints unneeded messages, especially during development and testing. In tests, I see the entire config printed to screen every time I create an app (each test), instead of the normal pytest `.....F..E.`
* It tires to pretty-print the config (for readability?), but it doesn't work - when running tests I see the config all on one line: `<Config {'JSON_AS_ASCII': True, 'FEATURE_EVAL_LOCATION': False, 'SESSION_REFRESH_EACH_REQUEST': True, 'DATASET_DIR': '/data/datasets', 'SENTRY_DSN': '' ... [lots more chopped]...>`

Reply:
Action points:
 - [ ] remove `config_path` argument
 - [ ] standardaise on consul_config.py _or_ config.py in both apps
 - [ ] only load config.py during development
 - [ ] only use `print` in production
 - [ ] correctly pretty-print config in production

Reply:
Commenting for posterity: https://github.com/metabrainz/acousticbrainz-server/pull/294, was reading this PR and this conversation looked unresolved.